src/Entity/Dossier.php

```diff
@@ -188,4 +188,16 @@ public function hasDocuments(): bool
     {
         return $this->documents->count() || $this->sousDossiers->exists(fn (int $key, Dossier $dossier) => $dossier->hasDocuments());
     }
+
+    public function toggleVisibility(): void
+    {
+        $this->setBPrive(!$this->getBPrive());
+
+        $subData = [...$this->getSousDossiers()->toArray(), ...$this->getDocuments()->toArray()];
+        foreach ($subData as $subDatum) {
+            if ($subDatum->getBPrive() !== $this->getBPrive()) {
+                $subDatum->toggleVisibility();
+            }
```

Comment:
We could directly use ->filter() on getSousDossiers() and getDocuments()  rather than a if condition in the foreach